framework/framework.go

```diff
@@ -35,56 +39,71 @@ type Config struct {
 	// can be versioned and different resources can be executed depending on the
 	// custom object being reconciled.
 	ResourceRouter func(ctx context.Context, obj interface{}) ([]Resource, error)
+	TPR            tpr.Interface
```

Comment:
I'd like to be able to use the framework for every kubernetes object. Not only custom objects inside TPRs.

Reply:
IMO you can do this. The TPR/CRD should maybe just be optional. We only use it to ensure its existence on boot. When you use the framework for pods you do not need to provide the TPR/CRD and can just go ahead without it. Or what was your thinking? Maybe suggest a solution.

Reply:
> When you use the framework for pods you do not need to provide the TPR/CRD and can just go ahead without it.

There is `invalidConfigError` returned when `config.TPR == nil` so I don't think so.

If you want to make that optional, fine. But please make sure CRD is handled as well and proper godoc is added.

Reply:
> The TPR/CRD should maybe just be optional.

Reply:
I made the TPR optional and added godoc. I stated TPR is deprecated. I would like to leave out the CRD magic for now since we are not using it internally and I do not know how it should be later on. I would like to more or less keep the copy/paste code and move forward. Lets please sort CRDs in a separate PR.

Reply:
Ack. I'm good with current state.